my_enums.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
+module Enumerable #:nodoc: all
+  def my_each
+    length.times do |n|
+      current = self[n]
+      yield(current)
+    end
+  end
+
+  def my_each_with_index
+    length.times do |n|
+      index = n
+      current = self[n]
+      yield(index, current)
+    end
+  end
```

Comment:
- [ ] enumerables my_each_with_index calls the given block once for each element in self
- [ ] enumerables my_each_with_index returns an enumerator if no block is given